src/vm.rb

```diff
@@ -10,8 +10,8 @@ class Vm
     attr_reader :node
 
     def initialize(node)
-      @libvirt ||= Libvirt.open("qemu://#{node.config.libvirt_host}/system")
       @node = node
+      @libvirt ||= certificate? ? libvirt : generate_certificate
```

Comment:
Now there's a `libvirt` method you could move the initializing of `@libvirt` to this rather than here (which will be possible with the change I suggest in the main review comment).